net.go

```diff
@@ -109,11 +122,11 @@ func DialTimeout(network, addr string, timeout time.Duration, ctx *Ctx,
 func DialSession(network, addr string, ctx *Ctx, flags DialFlags,
 	session []byte) (*Conn, error) {
 	var d net.Dialer
-	return dialSession(d, network, addr, ctx, flags, session)
+	return dialSession(d, context.Background(), network, addr, ctx, flags, session)
 }
 
-func dialSession(d net.Dialer, network, addr string, ctx *Ctx, flags DialFlags,
-	session []byte) (*Conn, error) {
+func dialSession(d net.Dialer, context context.Context, network, addr string,
+	ctx *Ctx, flags DialFlags, session []byte) (*Conn, error) {
```

Comment:
Let's stop adding unnecessary arguments here and reorganize the function a little.

For example, it could be something like this:

```suggestion
func createSession(c net.Conn, ctx *Ctx, flags DialFlags, session []byte) (*Conn, error) {
```

1. A separate function for `ctx *Ctx` preparation:
https://github.com/tarantool/go-openssl/blob/bdadb815fbb91fdee2b475f7f38b17a7f56db0e0/net.go#L134-L141

2. A separate function for `c net.Conn` creation:
https://github.com/tarantool/go-openssl/blob/bdadb815fbb91fdee2b475f7f38b17a7f56db0e0/net.go#L130-L133
and
https://github.com/tarantool/go-openssl/blob/bdadb815fbb91fdee2b475f7f38b17a7f56db0e0/net.go#L143-L146

or even
```suggestion
func setSession(client *Conn, flags DialFlags, session []byte) (*Conn, error) {
```

Reply:
Or you can split this function in some other way.

Reply:
Updated.